ui/dashboards/src/components/Variables/VariableEditorForm/variable-editor-form-model.ts

```diff
@@ -29,6 +30,7 @@ export function getInitialState(initialVariableDefinition: VariableDefinition) {
   if (initialVariableDefinition.kind === 'ListVariable') {
     listVariableFields.allowMultiple = initialVariableDefinition.spec.allow_all_value ?? false;
     listVariableFields.allowAll = initialVariableDefinition.spec.allow_all_value ?? false;
+    listVariableFields.capturing_regexp = initialVariableDefinition.spec.capturing_regexp ?? '';
```

Comment:
Currently as we are defining on the fly the struct `listVariableFields`, it doesn't seem you can specify the type of the field contains inside, since the type is defined based on the value you are putting there.

So if I'm putting an empty string, then the type of `listVariableFields.capturing_regexp` is a string. If I put `undefined` as a value, the type is then `undefined`. Excepting by defining an interface, I didn't find the way to say `listVariableFields.capturing_regexp` can be `string | undefined`

Reply:
I made a comment above on how to do that